server/utils/helpers/activityTimeSlotMappers.ts

```diff
@@ -344,6 +344,89 @@ export function calculateUniqueSlots(slotsA: Slot[], slotsB: Slot[]): Slot[] {
     .filter(s => s.daysOfWeek.length > 0)
 }
 
+export function calculateExclusionSlots(exclusions: Slot[], updatedExclusions: Slot[]): Slot[] {
+  const addedSlots: Slot[] = []
+
+  exclusions.forEach(exclusion => {
+    const filteredExclusions = updatedExclusions.filter(
+      e => e.timeSlot === exclusion.timeSlot && e.weekNumber === exclusion.weekNumber,
+    )
+    const days: ('MONDAY' | 'TUESDAY' | 'WEDNESDAY' | 'THURSDAY' | 'FRIDAY' | 'SATURDAY' | 'SUNDAY')[] =
+      exclusion.daysOfWeek
+
+    const daysToAdd: ('MONDAY' | 'TUESDAY' | 'WEDNESDAY' | 'THURSDAY' | 'FRIDAY' | 'SATURDAY' | 'SUNDAY')[] = []
+    days.forEach(day => {
+      const dayNotFound = filteredExclusions.filter(s => s.daysOfWeek.includes(day)).length === 0
+      if (dayNotFound) {
+        daysToAdd.push(day)
+      }
+    })
+    if (daysToAdd.length > 0) {
+      const slot: Slot = {
+        weekNumber: exclusion.weekNumber,
+        timeSlot: exclusion.timeSlot,
+        customStartTime: exclusion.customStartTime,
+        customEndTime: exclusion.customEndTime,
+        monday: daysToAdd.includes('MONDAY'),
```

Comment:
Maybe we could use the enum set up in activityTimeSlotMappers for the day strings?
IT's called DayOfWeekEnum

Reply:
will check

Reply:
updated